code/game/objects/structures/crates_lockers/closets/secure/craftable/freezer.dm

```diff
@@ -0,0 +1,204 @@
+		//you need to unlock to perform the operation else anyone can change access on a locked unit
+		if(card_reader_installed)
+			balloon_alert(user, "already installed!")
+			return TRUE
+
+		if(do_after(user, 4 SECONDS))
+			card_reader_installed = TRUE
+			attacking_item.moveToNullspace()
+			qdel(attacking_item)
+			balloon_alert(user, "card reader installed")
+
+		return TRUE
+	else if(istype(attacking_item, /obj/item/pen))
+		//you need to unlock to perform the operation else anyone can change name & description on a locked closet
+		if(locked)
+			balloon_alert(user, "unlock first!")
+			return TRUE
+
+		//you cant rename departmental lockers cause thats vandalism
+		if(access_type != PERSONAL)
+			balloon_alert(user, "not yours to rename!")
+			return TRUE
+
+		var/name_set = FALSE
+		var/desc_set = FALSE
+
+		var/str = tgui_input_text(user, "Personal Locker Name", "Locker Name")
+		if(!isnull(str))
+			name = str
+			name_set = TRUE
+
+		str = tgui_input_text(user, "Personal Locker Description", "Locker Description")
+		if(!isnull(str))
+			desc = str
+			desc_set = TRUE
+
+		var/bit_flag = NONE
+		if(name_set)
+			bit_flag |= UPDATE_NAME
+		if(desc_set)
+			bit_flag |= UPDATE_DESC
+		if(bit_flag != NONE)
+			update_appearance(bit_flag)
+
+		return TRUE
+	else if(card_reader_installed)
+		if(attacking_item.tool_behaviour == TOOL_CROWBAR)
+			if(!attacking_item.use_tool(src, user, 4 SECONDS))
+				return TRUE
+
+			new /obj/item/stock_parts/card_reader(drop_location())
+			card_reader_installed = FALSE
+			return TRUE
+
+		if(broken)
+			balloon_alert(user, "its broken!")
+			return TRUE
+		var/obj/item/card/id/id = null
+		if(istype(attacking_item, /obj/item/card/id))
+			id = attacking_item
+		else if(istype(attacking_item, /obj/item/modular_computer/pda))
+			var/obj/item/modular_computer/pda/pda = attacking_item
+			id = pda.computer_id_slot
+		if(isnull(id))
+			return ..()
+		else if(isnull(electronics))
+			balloon_alert(user, "missing electronics!")
+			return TRUE
+
+		//you need to unlock to perform the operation else anyone can change access on a locked closet
+		if(locked)
+			balloon_alert(user, "unlock first!")
+			return TRUE
+
+		if(access_locked)
+			balloon_alert(user, "access panel locked!")
+			return TRUE
+
+		//change the access type
+		var/static/list/choices = list(
+			"Personal" = PERSONAL,
+			"Departmental" = DEPARTMENTAL,
+			"None" = FREE_ACCESS
+		)
+		var/choice = tgui_input_list(user, "Set Access Type", "Access Type", choices)
+		if(isnull(choice))
+			return
+		access_type = choices[choice]
+
+		id_card = null
+		switch(access_type)
+			if(PERSONAL) //only the player who swiped their pda has access.
+				id_card = id
+			if(DEPARTMENTAL) //anyone who has the same access permissions as this id has access
+				electronics.accesses = id.GetAccess()
+				set_access()
+			if(FREE_ACCESS) //free for all
+				electronics.accesses = list()
+				set_access()
+		balloon_alert(user, "access is now [choice]")
```

Comment:
For every time you do this, is choice guaranteed to be lowercase? I don't think it is, so you should lowertext in this call. 

BTW nit: I think this should be `set to "[choice]"`, since the player already know they just used an access prompt

Reply:
Yup